### PR TITLE
[DOC] Enhance distance function related docs

### LINKED ISSUE #977: [DOC] MSM docs issue
### Describe the issue linked to the documentation

image

it should be \ldots in latex form
Also the source code link its broken, if u click it will redirect to this https://github.com/aeon-toolkit/aeon/blob/main/aeon/distances.py

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
this tidies up the docstrinbgs of distance based algorithms

1. In distances, adds a function to extract the names of distance functions. This is brittle, but I think it better than maintaining another structure to update to add a new distance
2. Rewords some doc strings of distance based things
3. Adds some testing to improve coverage
4. Begins the rehabilitation of MPDist
5. Removes an unused timing function in distances, restructures the distance utils
6. Added manhattan distance where missin
7. Tidied up all the distance docs
8. Fixes #977

### CODE REVIEW

File: aeon/distances/_msm.py [hunk 1]
```diff
@@ -355,12 +355,14 @@ def msm_pairwise_distance(
...
-            (m_instances, m_timepoints) or (m_timepoints,), default=None
-        A collection of time series instances.
+    X : np.ndarray
+        A collection of time series instances  of shape ``(n_instances, n_timepoints)``
+        or ``(n_instances, n_channels, n_timepoints)``.
+    y : np.ndarray or None, default=None
+        A single series or a collection of time series of shape ``(m_timepoints,)`` or
+        ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
+        calculated.
```
Comment (chrisholder) [inline aeon/distances/_msm.py:365]: msm pairwise distance

File: aeon/distances/_wddtw.py [hunk 2]
```diff
@@ -180,12 +180,14 @@ def wddtw_pairwise_distance(
...
-    y : np.ndarray, of shape (m_instances, m_channels, m_timepoints) or
-            (m_instances, m_timepoints) or (m_timepoints,), default=None
-        A collection of time series instances.
+    X : np.ndarray
+        A collection of time series instances  of shape ``(n_instances, n_timepoints)``
+        or ``(n_instances, n_channels, n_timepoints)``.
+    y : np.ndarray or None, default=None
+        A single series or a collection of time series of shape ``(m_timepoints,)`` or
+        ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
```
Comment (chrisholder) [inline aeon/distances/_wddtw.py:189]: wddtw pairwise distance

File: aeon/distances/_wdtw.py [hunk 3]
```diff
@@ -47,12 +47,14 @@ def wdtw_distance(
...
-    y : np.ndarray
-        Second time series, either univariate, shape ``(n_timepoints,)``, or
-        multivariate, shape ``(n_channels, n_timepoints)``.
+    X : np.ndarray
+        A collection of time series instances  of shape ``(n_instances, n_timepoints)``
+        or ``(n_instances, n_channels, n_timepoints)``.
+    y : np.ndarray or None, default=None
+        A single series or a collection of time series of shape ``(m_timepoints,)`` or
+        ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
```
Comment (chrisholder) [inline aeon/distances/_wdtw.py:56]: wdtw pairwise distance

**Review by chrisholder — CHANGES_REQUESTED** [review]

File: aeon/distances/_adtw.py [hunk 4]
```diff
@@ -214,6 +212,10 @@ def adtw_pairwise_distance(
     y : np.ndarray or None, default=None
         A single series or a collection of time series of shape ``(m_timepoints,)`` or
         ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
```
Comment (chrisholder) [inline aeon/distances/_adtw.py:215]: adtw pairwise distance

File: aeon/distances/_erp.py [hunk 5]
```diff
@@ -266,12 +266,14 @@ def erp_pairwise_distance(
...
-    y : np.ndarray, of shape (m_instances, m_channels, m_timepoints) or
-            (m_instances, m_timepoints) or (m_timepoints,), default=None
-        A collection of time series instances.
+    X : np.ndarray
+        A collection of time series instances  of shape ``(n_instances, n_timepoints)``
+        or ``(n_instances, n_channels, n_timepoints)``.
+    y : np.ndarray or None, default=None
+        A single series or a collection of time series of shape ``(m_timepoints,)`` or
+        ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
```
Comment (chrisholder) [inline aeon/distances/_erp.py:275]: erp pairwise distance

File: aeon/distances/_edr.py [hunk 6]
```diff
@@ -346,10 +348,10 @@ def edr_alignment_path(
...
-        First time series.
-    y : np.ndarray, of shape (m_channels, m_timepoints) or (m_timepoints,)
-        Second time series.
+    x : np.ndarray
+        First time series, shape ``(n_channels, n_timepoints)`` or ``(n_timepoints,)``.
+    y : np.ndarray
+        Second time series, shape ``(m_channels, m_timepoints)`` or ``(m_timepoints,)``.
     window : float, default=None
         The window to use for the bounding matrix. If None, no bounding matrix
         is used.
```
Comment (chrisholder) [inline aeon/distances/_edr.py:357]: No change to edr pairwise docstring?

**Review by chrisholder — CHANGES_REQUESTED** [review]

**Comment by chrisholder** [comment]
Just a note you updated the pairwise distances docstring but didn't include edr, twe

File: aeon/distances/_lcss.py [hunk 7]
```diff
@@ -228,12 +228,14 @@ def lcss_pairwise_distance(
...
-    y : np.ndarray, of shape (m_instances, m_channels, m_timepoints) or
-            (m_instances, m_timepoints) or (m_timepoints,), default=None
-        A collection of time series instances
+    X : np.ndarray
+        A collection of time series instances  of shape ``(n_instances, n_timepoints)``
+        or ``(n_instances, n_channels, n_timepoints)``.
+    y : np.ndarray or None, default=None
+        A single series or a collection of time series of shape ``(m_timepoints,)`` or
+        ``(m_instances, m_timepoints)`` or ``(m_instances, m_channels, m_timepoints)``.
+        If None, then the squared pairwise distance between the instances of X is
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/distances/_lcss.py:237]: too much cut and pasting!

**Comment by TonyBagnall** [comment]
> Just a note you updated the pairwise distances docstring but didn't include edr, twe

sorted I think, thanks

**Pull request merged**